src/sentry/static/sentry/app/utils/withProject.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import SentryTypes from 'app/sentryTypes';
+import getDisplayName from 'app/utils/getDisplayName';
+
+/**
+ * Currently wraps component with project from context
+ */
+const withProject = WrappedComponent =>
+  class extends React.Component {
+    static displayName = `withProject(${getDisplayName(WrappedComponent)})`;
+    static contextTypes = {
```

Comment:
Should we be adding new usage of the legacy context API? Could the active project come from a redux store, or use a projectId to get the project out of the existing ProjectsStore?

Reply:
This is specifically for use with the `ProjectContext` parent component. The idea is that using a HoC like this decouples components that use it from the context API, and we can later replace the context API here with something else